source/fundamentals/authentication.txt

```diff
@@ -142,6 +143,180 @@ string as follow:
+The next examples show how to provide your credentials by 
+setting environment variables for the following types of authentication:
+
+* Programmatic access keys
+* ECS container credentials
+* EC2 container credentials
+* EKS container credentials
+
+.. important::
+
+   Before checking environment variables for AWS IAM credentials, the {+driver-short+} 
+   will look for credentials in the following places:
+
+   * App.config and web.config files for the project
+   * Web identity provider
+   * Local AWS configuration
+
+The following examples show how you can set your **programmatic access keys**
+in environment variables by using ``bash`` or a similar shell:
+
+.. code-block:: bash
+
+   export AWS_ACCESS_KEY_ID=<awsKeyId>
+   export AWS_SECRET_ACCESS_KEY=<awsSecretKey>
+   export AWS_SESSION_TOKEN=<awsSessionToken>
+
+.. note::
+   
+   Omit the line containing ``AWS_SESSION_TOKEN`` if you don't need an AWS
+   session token for that role.
+
+To authenticate by using **ECS container credentials**, set the ECS-endpoint-relative 
+URI in an environment variable by using ``bash``
+or a similar shell as shown in the following example:
+
+.. code-block:: bash
+
+   export AWS_CONTAINER_CREDENTIALS_RELATIVE_URI=<your ECS endpoint>
+
+To authenticate by using **EC2 container credentials**, make sure none of the
+aforementioned environment variables are set. The driver obtains the
+credentials from the default IPv4 EC2 instance metadata endpoint.   
+
+To authenticate by using **EKS cluster credentials**,
+create a file that contains your OpenID Connect (OIDC) token, then set the path to 
+this file in an environment variable by using ``bash`` or a similar shell as shown in
+the following example:
+
+.. code-block:: bash
+
+   export AWS_WEB_IDENTITY_TOKEN_FILE=<path to file containing your OIDC token>
+
+Specify Your Credentials in MongoClientSettings
++++++++++++++++++++++++++++++++++++++++++++++++
+
+You can supply your AWS IAM credentials on a ``MongoClientSettings`` object either by 
+using a ``MongoCredential`` instance or as part of the connection string.
+
+To use ``MongoClientSettings`` for ``MONGODB-AWS`` authentication, 
+you must perform the following steps:
+
+1. Specify the authentication mechanism and authentication source
+#. Supply the credentials
+
+To specify the authentication mechanism and authentication source by using a
+``MongoCredential``, create a ``MongoCredential`` instance and add it to your
+``MongoClientSettings`` as shown in the following example:
+
+.. code-block:: csharp
+
+   var awsCredentials = new MongoCredential("MONGODB-AWS", new MongoExternalIdentity("<awsKeyId>"), new PasswordEvidence("<awsSecretKey>"));
+
+   var mongoClientSettings = MongoClientSettings.FromConnectionString("mongodb+srv://<atlasUri>");
+   mongoClientSettings.Credential = awsCredentials;
+   var client = new MongoClient(mongoClientSettings);
+
+.. note:: AWS Session Token in MongoCredential
```

Comment:
Suggestion:

I think AWS session token is a common enough use case to put in normal documentation rather than an admonition. I think moving the content outside of the admonition would work. Seems like the session token was the default case in the Environment Variables examples, but here it's the exception case.

Same suggestion applies to the admonition starting on line 309.